crates/stages/src/stages/execution.rs

```diff
@@ -121,6 +123,10 @@ impl<EF: ExecutorFactory> ExecutionStage<EF> {
         let max_block = input.target();
         let prune_modes = self.adjust_prune_modes(provider, start_block, max_block)?;
 
+        // We only append receipts to static files if there is no pruning or filtering configured
+        // for receipts.
+        let snapshotter = try_prepare_snapshotter(&self.prune_modes, provider, start_block)?;
```

Comment:
this function is standalone for testing reasons?

then this makes sense, though I'd also add a fn `try_prepare_snapshotter(&self` if possible

Reply:
i guess this was doing too much and not very clear
* check if `snapshotter` is to be used. - put it outside of the function
* consistency check of the static file data (eg. do i have more receipts than database expects? then unwind static files)

added more docs that hopefully clarify